service-api/module/Application/src/ControllerFactory/AuthControllerAbstractFactory.php

```diff
@@ -62,6 +63,12 @@ public function __invoke(ContainerInterface $container, $requestedName, array $o
 
         $controller = new $requestedName($authenticationService, $service);
 
+        $traitsUsed = class_uses($controller);
```

Comment:
Might be worth adding a comment about how 'class_uses' does *not* return traits in parent classes. Works here but we did have an issue with that.